src/jsonrpc/types.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+/* line: 1-based, column: 0-based */
+export type Position = {
+  line: number; // 1-based
+  column: number; // 0-based
+} | { path?: JsonPath };
+
+export type JsonPath = (string | number)[];
+
+export interface SourceLocation {
+  document: string;
+  Position: Position;
+}
+
+export interface Message {
+  Channel: "information" | "warning" | "error" | "debug" | "verbose";
```

Comment:
I think they generally are named as `info` / `warn` not sure why do we need them to be complete word. Not sure may be autorest requires it?